core/common/src/Instant.kt

```diff
@@ -181,6 +181,16 @@ public val Instant.isDistantFuture: Boolean
  */
 public fun String.toInstant(): Instant = Instant.parse(this)
 
+/**
+ * Converts this string representing an instant in ISO-8601 format including date and time components and
+ * the time zone offset to an [Instant] value.
+ * 
+ * Returns null, if this string does not match the ISO-8601 format or the boundaries of [Instant] are exceeded.
+ *
+ * See [Instant.parse] for examples of instant string representations.
+ */
+public fun String.toInstantOrNull(): Instant? = try { Instant.parse(this) } catch(cause: IllegalArgumentException) { null }
```

Comment:
You could use [runCatching](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/run-catching.html) here unless you only want to explicitly catch `IllegalArgumentException`? It is available for multiplatform since Kotlin 1.3.

```suggestion
public fun String.toInstantOrNull(): Instant? = runCatching { Instant.parse(this) }.getOrNull()
```

Reply:
I do want to only catch `IllegalArgumentException` because this exception is thrown by `java.time.Instant` when passing an invalid instant. Also runCatching creates a new class, `Result`, just to do the same thing. But I guess, it is not merged, because ideally it would be better to not use catch at all because it is still inefficient. Maybe checking the format with a regex first would be better.

Reply:
In that case let's keep the `try ... catch`. I doubt creating an instance of `Regex` and executing that regex is any more performant than `try ... catch` though.